app/src/main/java/com/kabouzeid/gramophone/adapter/CategoryAdapter.java

```diff
@@ -0,0 +1,204 @@
+package com.kabouzeid.gramophone.adapter;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.ColorFilter;
+import android.graphics.Paint;
+import android.graphics.PixelFormat;
+import android.graphics.Rect;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.GradientDrawable;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.graphics.drawable.DrawableCompat;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.helper.ItemTouchHelper;
+import android.view.LayoutInflater;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.TextView;
+
+import com.kabouzeid.appthemehelper.ThemeStore;
+import com.kabouzeid.gramophone.R;
+import com.kabouzeid.gramophone.util.SwipeAndDragHelper;
+import com.kabouzeid.gramophone.model.Category;
+
+import java.util.ArrayList;
+
+public class CategoryAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> implements SwipeAndDragHelper.ActionCompletionContract {
+    private ArrayList<Category> categories;
+    private ItemTouchHelper touchHelper;
+    private ColorStateList color;
+
+    public CategoryAdapter(ArrayList<Category> categories, ColorStateList color) {
+        this.categories = copy(categories);
+        this.color = color;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.preference_dialog_library_categories_listitem, parent, false);
+        return new CategoryViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        Category category = categories.get(position);
+        CategoryViewHolder h = (CategoryViewHolder) holder;
+        h.checkBox.setChecked(category.visible);
+        h.title.setText(h.title.getResources().getString(category.id.key));
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
+            h.checkBox.setButtonTintList(color);
+        } else {
+            Drawable checkDrawable =
+                    ContextCompat.getDrawable(h.checkBox.getContext(), R.drawable.abc_btn_check_material);
+            Drawable drawable = DrawableCompat.wrap(checkDrawable);
+            DrawableCompat.setTintList(drawable, color);
+            h.checkBox.setButtonDrawable(drawable);
+        }
+
+        h.checkBox.setOnClickListener((view) -> {
+                h.checkBox.setChecked(category.visible = !category.visible);
+                notifyDataSetChanged();
```

Comment:
This is why the animation is missing. As soon as the animation should start onBindViewHolder() will be called again.

Reply:
So the checkboxes are animated by default? That would be nice.

material-dialogs also shows a different background on selection which improves the user experience.